Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -929,7 +929,54 @@ private void DestroyAllServices()
         {
             if (!ExecuteOnAllServices(service => service.Destroy())) { return; }
 
+            List<Type> serviceTypes = activeSystems.Keys.ToList<Type>();
+            foreach (Type type in serviceTypes)
+            {
+                if (typeof(IMixedRealityBoundarySystem).IsAssignableFrom(type))
```

Comment:
>IsAssignableFrom [](start = 56, length = 16)

Are we looking for assignable, or identity? i.e. UnregisterService<IMixedRealityBoundarySystem> looks like it takes exactly that interface, not just some type that happens to extend/be related to it.

Reply:
assignable. just in case a developer extends the interface

Reply:
In that case I don't fully understand how this works - we'll then look to unregister a IMixedRealityBoundarySystem object by looking at the IMixedRealityBoundarySystem type array in the list of services, but if they've registered it as a subtype it won't be there.

I guess the question is, if we have the list of types and values (i.e. the service instances, which I assume are in activeSystems) why not just call UnregisterService passing the service instance directly? It seems a bit roundabout to have to make that function re-find the object we already have here.

---
In reply to: [288762686](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4619#discussion_r288762686) [](ancestors = 288762686)

Reply:
good point!

Reply:
> I guess the question is, if we have the list of types and values (i.e. the service instances, which I assume are in activeSystems)

The issue is that to properly call the generic <T> method, we need to know T at compile time.

Reply:
> In that case I don't fully understand how this works - we'll then look to unregister a IMixedRealityBoundarySystem object 

On further reflection, for a system to be a core system, it must implement one of the core interfaces (in your example IMixedRealityBoundarySystem). If it is registered any other way, it would not be a core service.